src/components/Inputs/Input.types.ts

```diff
@@ -323,7 +351,7 @@ export interface InputProps<T>
    * input readonly.
    * @default false
    */
-  readonly?: boolean;
+  readonly?: boolean | ReadOnlyProps;
```

Comment:
I did this review in reverse order, so this is called out below as well. Can we separate the props and the boolean? It would read nicer without the type checking, etc